agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/BatchSpanProcessor.java

```diff
@@ -206,8 +216,12 @@ private void flush() {
         }
       }
       exportCurrentBatch();
-      flushRequested.get().succeed();
-      flushRequested.set(null);
+      CompletableResultCode.ofAll(pendingExports).join(exporterTimeoutNanos, TimeUnit.NANOSECONDS);
+      CompletableResultCode flushResult = flushRequested.get();
+      if (flushResult != null) {
```

Comment:
is null check needed?

Reply:
I pulled this change in from upstream

Reply:
maybe upstream needs to be updated? based on the old code, null check doesn't seem to be necessary.

Reply:
it looks like upstream added the null check to make for nullaway static analysis happy, I'd rather stay in sync with them